[BugFix] db lock deadlock cause by isRestoreReplica function in tabletReport(#32789)

srlch opened the pull request:
tabletInvertedIndex lock and db lock must lock and unlock in the following order： lock: db->tabletInvertedIndex
unlock:tabletInvertedIndex -> db

The problem here is that tabletInvertedIndex has been locked so it must under the db lock. isRestoreReplica lock db lock again the cause the deadlock

Fixes #32789

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|4|5|80.00%|[544]|

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

srlch committed:
[BugFix] db lock deadlock cause by isRestoreReplica function in tabletReport(#32789)

tabletInvertedIndex lock and db lock must lock and unlock in the following order：
lock: db->tabletInvertedIndex
unlock:tabletInvertedIndex -> db

The problem here is that tabletInvertedIndex has been locked so it must under the db lock.
isRestoreReplica lock db lock again the cause the deadlock

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

wanpengfei-git commented:
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|1|1|100.00%|[]|
